core/invoker/src/main/scala/whisk/core/containerpool/ContainerPool.scala

```diff
@@ -120,11 +120,7 @@ class ContainerPool(
         // Container got removed
         case ContainerRemoved =>
             freePool.remove(sender())
-            busyPool.remove(sender())
-
-        // Activation completed
-        case ActivationCompleted =>
-            feed ! ContainerReleased
+            busyPool.remove(sender()).foreach(_ => feed ! ContainerReleased)
```

Comment:
so where previously the slow was released on completion, and a new request may be received when a container isn't available, instead, indicate capacity availability when the container pool can actually readily allocate a new slot - is that correct?